clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -755,7 +755,21 @@ private void dropPartitionSynchronously(PubSubTopicPartition topicPartition) {
   }
 
   public boolean hasAnySubscription() {
-    return !partitionConsumptionStateMap.isEmpty();
+    return !partitionConsumptionStateMap.isEmpty() || hasAnyPendingSubscription();
+  }
+
+  /**
+   * Check the consumer action queue to see if there are any pending SUBSCRIBE actions.
+   */
+  public boolean hasAnyPendingSubscription() {
+    return consumerActionsQueue.stream().anyMatch(action -> action.getType() == SUBSCRIBE);
+  }
+
+  /**
+   * This helper function will check if the ingestion task has been idle for a long time.
+   */
+  public boolean isIdleOverThreshold() {
```

Comment:
So we reset this counter once it reaches maxIdleCounter. when it will be > maxIdleCounter ?

Reply:
There is a "return" at the end of the condition block that handle idle SIT. counter is not reset to 0. Otherwise, the new tests wouldn't work anyway.